### PR TITLE
UI: run set_brightness in QtConcurrent

### PR DESCRIPTION
**Pull request by deanlee**
~use std::future to control the life time of the thread and avoid accidentally starting multiple threads to set brightness.~
move to thread

### CODE REVIEW

**Comment by adeebshihadeh** [comment]
Generally, there's a simpler way to write a PR that looks like "XYZ: use more C++". In this case, why don't we just spin up another thread that sets the brightness at like 10Hz like we used to pre-Qt? A simple thread that sets the brightness and sleeps in a loop is way simpler than: 
``` c++
    if (!brightness_future.valid() || brightness_future.wait_for(0ms) == std::future_status::ready) {
      brightness_future = std::async(std::launch::async, Hardware::set_brightness, brightness);
      last_brightness = brightness;
    }
```

**Comment by deanlee** [comment]
use QThread to facilitate moving updateWakefulness to thread if needed.

**Comment by adeebshihadeh** [comment]
This is a massive diff to cleanup the original stated issue of "multiple brightness threads can be created". Has the goal changed? If not, why move the whole device update to a thread? If this is merged as is, I can easily see a future PR to refactor this again and "simplify" everything into a single thread, a la [Chesterton's Fence](https://wiki.lesswrong.com/wiki/Chesterton%27s_Fence).

**Comment by deanlee** [comment]
There may be no simple way to spin up another thread to update brightness because SubMaster and UIState are not thread safe.
It may be better to use QFuture and QtCurrent at present, this can avoid starting multiple threads,and eliminate the overhead of creating thread

File: selfdrive/ui/ui.cc [hunk 1]
```diff
@@ -255,7 +257,9 @@ Device::Device(QObject *parent) : brightness_filter(BACKLIGHT_OFFROAD, BACKLIGHT
 }
 
 void Device::update(const UIState &s) {
-  updateBrightness(s);
+  if (s.sm->frame % 2 == 0) {
```
Comment (pd0wm) [inline selfdrive/ui/ui.cc:260]: Why was the frequency changed? This messes with the `brightness_filter` time constant.

**Comment by pd0wm** [comment]
If you revert the panda changes I'll merge.

**Comment by deanlee** [comment]
rebased

**Pull request merged**